xbmc/addons/AddonRepos.h

```diff
@@ -0,0 +1,188 @@
+/*
+ *  Copyright (C) 2005-2020 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include "AddonDatabase.h"
+
+#include <map>
+#include <vector>
+
+namespace ADDON
+{
+/**
+ * Struct - RepoInfo
+ */
+struct RepoInfo
+{
+  std::string m_repoId;
+  std::string m_origin;
+};
+
+class CAddonMgr;
+class CRepository;
+class IAddon;
+
+/**
+ * Class - CAddonRepos
```

Comment:
Can you please detail what the responsibility of this class is so that it won't get cluttered/abused in the future?

Reply:
sure